tedana/selection/tedica.py

```diff
@@ -386,10 +386,10 @@ def kundu_selection_v2(comptable, n_echos, n_vols):
         unclf = np.setdiff1d(unclf, midk)
 
         # Find components to ignore
-        # Ignore high variance explained, poor decision tree scored components
-        new_varex_lower = stats.scoreatpercentile(
-            comptable.loc[unclf[:num_acc_guess], "variance explained"], LOW_PERC
-        )
+        # Of the remaining components, ignore ones with higher variance even if their
+        # decision tree scores are poor
```

Comment:
```suggestion
        # Of the remaining components, ignore ones with higher variance even if their
        # decision tree scores are poor, instead of rejecting them
```

Thanks for improving the comment. I think making it clear that they would otherwise be rejected will also help. However, I'm also confused. `candart` (the artifact candidate components) are never classified as rejected, so won't all of the un-ignored components from this step actually be accepted?

Reply:
Thank you for opening a can of worms. :)
This is a bit confusing, so I decided to go back to the original MEICA to explain better & it turns out there's another divergence in the original port.
To first answer your question, `candart` is used in both versions and is simply a bad variable name. It should be `candign` since these are candidates to ignore.
Now the difference. In the MEICA code, all the steps we now label I009-I012 were done on the same subset of components.
https://github.com/ME-ICA/me-ica/blob/53191a7e8838788acf837fdf7cb3026efadf49ac/meica.libs/select_model.py#L344-L360
`ncl` (now `unclf`) is assigned before this section and isn't changed until the end of the section. In tedana Main, after each classification step, we see:
https://github.com/ME-ICA/tedana/blob/f00cb25152142611b8e289ab59c7b5b8ab6eaf08/tedana/selection/tedica.py#L375

In practice, what this means is that the components rejected with I009 or I010 are NOT candidates to be ignored in I011 and I012 while they can change from rejected (actually midk) to ignored in the original MEICA. This vaguely makes sense since the goal of these final steps is to retain a few things that might have otherwise been rejected. That said, I011 and I012 operate on all remaining unclassified components so they can and do change components from `accepted` to `ignored`. All four of these steps are classifying based on different thresholds for `variance explained` & `d_table_score_scrub` so it would take a bit more thinking to see how they all interact.

That all said, there is the question about what to do now. This is the version of `tedana` people have been running for years. Do we want to:
1. Add a real breaking change into Main and then try to match that version in DTM?
2. Keep Main as is and (possibly) add another decision tree into DTM that will match the original MEICA?

I spoke to @jbteves and I lean towards (2) since breaking Main just before the DTM merge might cause unnecessary headaches. I still think it's worth merging this PR since it's correcting something that clearly wrong - Treating an unsorted array as if it was sorted. The difference between MEICA and tedana Main is a difference between pipelines where neither is inherently right.